__tests__/ppu_test.re

```diff
@@ -0,0 +1,162 @@
+open Jest;
+open Expect;
+open Spec;
+
+let nestest = rom("nestest");
+
+let ctrl_test_helper = (regs: Ppu.registers, n, reader, unset, set) => {
+  test("checking CTRL " ++ string_of_int(n) ++ " bit set", () => {
+    regs.control = 0;
+    expect(reader(regs)) |> toEqual(unset);
+  });
+  test("checking CTRL " ++ string_of_int(n) ++ " bit set", () => {
+    regs.control = 1 lsl n;
+    expect(reader(regs)) |> toEqual(set);
+  });
+};
+
+let mask_test_helper = (regs: Ppu.registers, n, reader) => {
+  test("checking MASK " ++ string_of_int(n) ++ " bit set", () => {
+    regs.mask = 0;
+    expect(reader(regs)) |> toEqual(false);
+  });
+  test("checking MASK " ++ string_of_int(n) ++ " bit set", () => {
+    regs.mask = 1 lsl n;
+    expect(reader(regs)) |> toEqual(true);
+  });
+};
+
+describe("PPU", () => {
+  let ppu = Ppu.build(nestest);
+  let regs = ppu.registers;
+
+  describe("PPUCTRL", () => {
+    ctrl_test_helper(regs, 0, Ppu.x_scroll_offset, 0, 256);
```

Comment:
I think it'd be slightly easier to scan if this helper were defined inside the `decribe("PPUCTRL", { ... })` block. FWIW, you could close over `regs` in that case. (Sim. for `PPUMASK` / `mask_test_helper` below)

Reply:
That's a good note. I'll reshuffle before merging!